### PR TITLE
Protcon pipeline clean 

### PR DESCRIPTION
**Pull request by jupinator**
This is a cleaner version of the Protcon pipeline. All comments make by JM were addressed  (except [one](https://github.com/GEO-BON/biab-2.0/pull/77#discussion_r1318883228) ).  I think is better to accept this commit and ignored this [one](https://github.com/GEO-BON/biab-2.0/pull/77).

### CODE REVIEW

**Comment by frousseu** [comment]
For the time interval in months to work, it has to be "4m" and not "4M" otherwise the string is not recognized.

File: scripts/studyarea_to_WKT/studyarea_to_WKT.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ - name: Victor Julio Rincon
+external_link: https://github.com/GEO-BON/biab-2.0
+inputs:
+  studyarea_path:
+    label: study area geometry
+    description: Text string representing the file path to the spatial file containing the spatial boundaries pertaining to the study area. Accepted file types include shapefile (.shp), raster (.tif/GeoTIFF) or  GeoPackage (.gpkg). This path specifies the complete path location of the file within the file system and is utilized for identifying and accessing the spatial boundaries contained within the file.
+    type: text
+    example: "/scripts/studyarea_to_WKT/input/Arauca.gpkg"
+  studyarea_epsg:
+    label: study area EPSG in
```
Comment (tpoisot) [inline scripts/studyarea_to_WKT/studyarea_to_WKT.yml:15]: "EPSG of the study area"?

File: scripts/spatial_to_DistanceMatrix/spatial_to_DistanceMatrix.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    label: spatial unit 
+    description: Refers to the column in the spatialunits_studyarea_data that denotes the spatial entity or geographic unit used for analysis. This column contains information about the spatial units that have been grouped based on the levels provided in the column_group input.
+    type: text
+  column_area:
+    label: column area
+    description: Refers to the column in the dataset that contains the estimated area for each of the spatial units under analysis. The estimation is expressed in square units (m2 or km2), as defined by the input unit_distance.
+    type: text
+  area_study_area:
+    label: total study area
+    description: Refers to the total area of the analyzed study area. The estimation of this area is expressed in square units (m2 or km2) as defined by the input unit_distance.
```
Comment (tpoisot) [inline scripts/spatial_to_DistanceMatrix/spatial_to_DistanceMatrix.yml:64]: the input is `distance_unit`, not `unit_distance`

File: scripts/spatial_to_DistanceMatrix/spatial_to_DistanceMatrix.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    label: unit for distance measurement
+    description: String value that defines the measurement unit for calculating distances between spatial units (m or km) and estimating their areas (m2 or km2) in the study area.
+    type: options
+    options:
+      - m
+      - km
+    example: "m"
+  spatial_file:
+    label: path to spatial file
+    description: String path represents the file location of a spatial file containing spatial units, which is used to calculate the distance matrix
```
[the remark this answers is not in the document]
Comment (tpoisot) [reply scripts/spatial_to_DistanceMatrix/spatial_to_DistanceMatrix.yml:29]: I would add a bit more context here, I agree it's not obvious what this one does

File: scripts/protcon_analyisis/protcon_analyisis.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    example: null
+  time_end:
+    label: end time 
+    description: Refers to the ending date of the time_interval. If this value is null or not provided, the minimum date from the column_date will be automatically taken as the starting date for the analysis
+    type: text
+    example: null
+outputs:
+  protcon_result:
+    label: protcon results
+    description: "Contains the results of the connectivity analysis between the spatial units within the study area. This table provides valuable information about the spatial relationships and connectivity patterns among different spatial entities"
```
Comment (tpoisot) [inline scripts/protcon_analyisis/protcon_analyisis.yml:71]: Is there enough room to give a brief overview of what the format of the data is, i.e. what are the columns in the CSV?

File: scripts/protcon_analyisis/protcon_analyisis.yml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    type: text/csv
+    example: "/scripts/protcon_analyisis/input/mtx_distance_centroid.csv"
+  area_study_area:
+    label: total study area
+    description: Refers to the total area of the analyzed study area. The unit of area in which it is defined is explicitly specified as the square of the unit_distance input (m2 or km2). 
+    type: int
+    example: 11000000000
+  distance_analysis:
+    label: distance analysis threshold
+    description: Refers to the threshold distance used to estimate if the areas are connected in a spatial analysis. This threshold value defines the maximum distance allowed between two spatial units for them to be considered as connected or neighboring. he unit of distance in which it is defined is explicitly specified in the unit_distance input (m or km).
```
Comment (tpoisot) [inline scripts/protcon_analyisis/protcon_analyisis.yml:22]: "he unit of" -> "The unit of"

File: scripts/protcon_analyisis/protcon_analyisis.yml [hunk 6]
```diff
@@ -0,0 +1,72 @@
+script: protcon_analyisis.R
+name: Connectivity of PAs
+author: 
+ - name: Maria Alejandra Molina Berbeo
```
Comment (tpoisot) [inline scripts/protcon_analyisis/protcon_analyisis.yml:4]: email missing

File: scripts/protcon_analyisis/protcon_analyisis.yml [hunk 7]
```diff
@@ -0,0 +1,72 @@
+script: protcon_analyisis.R
+name: Connectivity of PAs
```
Comment (tpoisot) [inline scripts/protcon_analyisis/protcon_analyisis.yml:2]: "PAs" -> "protected areas"

File: scripts/protcon_analyisis/protcon_analyisis.R [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+    dist_ids_pa_datav2<-   if(nrow(dist_ids_pa_data)>0){
+      dist_ids_pa_data[!duplicated(lapply(seq(nrow(dist_ids_pa_data)), function(j) dist_ids_pa_data[j,1:2]  )),]
+    } else {dist_ids_pa_data}
+    
+    
+    #Filter only the protected areas that are at a distances less than user distance
+    dist_ids_pa_con =dist_ids_pa_datav2[dist_ids_pa_datav2$values<distance, ]
+    
+    #Extract the unique id?s of protected areas at a distances less than user distance
+    pa_union =union(unique(dist_ids_pa_con$row),unique(dist_ids_pa_con$vars))
```
Comment (tpoisot) [inline scripts/protcon_analyisis/protcon_analyisis.R:140]: `pa_union =union`-> `pa_union = union`

File: scripts/protcon_analyisis/protcon_analyisis.yml [hunk 9]
```diff
@@ -0,0 +1,72 @@
+script: protcon_analyisis.R
```
Comment (tpoisot) [inline scripts/protcon_analyisis/protcon_analyisis.yml:1]: There is a typo in the file name, this should be changed

File: pipelines/Protcon-pipeline/Protcon-pipeline.md
Comment (tpoisot) [inline pipelines/Protcon-pipeline/Protcon-pipeline.md:1]: I really like the depth of this document, this is really clear

File: pipelines/Protcon-pipeline/Protcon-pipeline.Rmd [hunk 10]
```diff
@@ -0,0 +1,180 @@
...
+
+This methodology is based on a minimum distance in which protected areas are considered to be well connected; of course, this distance may vary according to the objective for which connectivity is desired. Once the protected areas that are at a distance equal to or less than the defined distance have been identified, the percentage that these represent with respect to the total protected areas of the total area of analysis (protcon) is estimated. To extract the percentage of protected areas that are not well connected (protuncon), the protcon is subtracted from the total number of protected areas in the area of interest. The indicator also generates the percentage values of protected (protected) and unprotected (unprotected) area of interest.
+
+$$
+protcon=\frac{A_{pac}}{A_{tpa}}*100
+$$
+**Where:**
```
Comment (tpoisot) [inline pipelines/Protcon-pipeline/Protcon-pipeline.Rmd:30]: leave a blank line after `$$` just in case

**Review by tpoisot — CHANGES_REQUESTED** [review]

File: scripts/protcon_analyisis/protcon_analyisis.R [hunk 11]
```diff
@@ -0,0 +1,193 @@
+#### Load required packages - libraries to run the script ####
+
+# Install necessary libraries - packages  
+packagesPrev<- installed.packages()[,"Package"] # Check and get a list of installed packages in this machine and R version
+packagesNeed<- list("magrittr", "terra", "raster", "sf", "fasterize", "pbapply", "this.path", "rjson") # Define the list of required packages to run the script
```
[the remark this answers is not in the document]
Comment (jupinator) [reply scripts/protcon_analyisis/protcon_analyisis.R:5]: Done

File: scripts/spatial_to_DistanceMatrix/spatial_to_DistanceMatrix.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jupinator) [reply scripts/spatial_to_DistanceMatrix/spatial_to_DistanceMatrix.yml:29]: done

File: scripts/spatial_to_DistanceMatrix/spatial_to_DistanceMatrix.yml [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+    type: int
+  area_unit:
+    label: unit for area measurement
+    description: Refers to the unit of measurement used for representing area in square units. The specific unit, whether square meters (m2) or square kilometers (km2), is determined by the input value provided for unit_distance.
+    type: options
+    options:
+      - m2
+      - km2
+  distance_unit:
+    label: unit for distance measurement
```
[the remark this answers is not in the document]
Comment (jupinator) [reply scripts/spatial_to_DistanceMatrix/spatial_to_DistanceMatrix.yml:74]: done

File: scripts/studyarea_to_WKT/studyarea_to_WKT.yml [hunk 13]
```diff
@@ -0,0 +1,31 @@
+script: studyarea_to_WKT.R
+name: study area geometry to WKT
```
[the remark this answers is not in the document]
Comment (jupinator) [reply scripts/studyarea_to_WKT/studyarea_to_WKT.yml:2]: ok, done

File: scripts/studyarea_to_WKT/studyarea_to_WKT.yml [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+description: This script converts the geometry of the study area into a well-known text (WKT) representation.
+author: 
+ - name: Maria Alejandra Molina Berbeo (mmolina@humboldt.org.co)
+ - name: Victor Julio Rincon
+external_link: https://github.com/GEO-BON/biab-2.0
```
[the remark this answers is not in the document]
Comment (jupinator) [reply scripts/studyarea_to_WKT/studyarea_to_WKT.yml:7]: done

File: scripts/studyarea_to_WKT/studyarea_to_WKT.yml [hunk 15]
```diff
@@ -0,0 +1,31 @@
...
+inputs:
+  studyarea_path:
+    label: study area geometry
+    description: Text string representing the file path to the spatial file containing the spatial boundaries pertaining to the study area. Accepted file types include shapefile (.shp), raster (.tif/GeoTIFF) or  GeoPackage (.gpkg). This path specifies the complete path location of the file within the file system and is utilized for identifying and accessing the spatial boundaries contained within the file.
+    type: text
+    example: "/scripts/studyarea_to_WKT/input/Arauca.gpkg"
```
[the remark this answers is not in the document]
Comment (jupinator) [reply scripts/studyarea_to_WKT/studyarea_to_WKT.yml:13]: done

File: scripts/studyarea_to_WKT/studyarea_to_WKT.yml (hunk 1 above)
Comment (jupinator) [reply scripts/studyarea_to_WKT/studyarea_to_WKT.yml:15]: i believe that the YML don't accept capital letters  in the first letter, so de description will be "study area EPSG"

File: scripts/spatial_to_DistanceMatrix/spatial_to_DistanceMatrix.yml (hunk 2 above)
Comment (jupinator) [reply scripts/spatial_to_DistanceMatrix/spatial_to_DistanceMatrix.yml:64]: done

File: scripts/studyarea_to_WKT/studyarea_to_WKT.yml (hunk 1 above)
Comment (jmlord) [reply scripts/studyarea_to_WKT/studyarea_to_WKT.yml:15]: Hi,
They do accept capital letters for labels. So you can turn it around the way you like :)
It is for the input and output key that snake case is needed. (studyarea_epsg in the example above)

Comment (jupinator) [reply scripts/studyarea_to_WKT/studyarea_to_WKT.yml:15]: done

File: scripts/protcon_analyisis/protcon_analyisis.yml (hunk 9 above)
Comment (jupinator) [reply scripts/protcon_analyisis/protcon_analyisis.yml:1]: done

File: scripts/protcon_analyisis/protcon_analyisis.yml (hunk 6 above)
Comment (jupinator) [reply scripts/protcon_analyisis/protcon_analyisis.yml:4]: done

File: scripts/protcon_analyisis/protcon_analyisis.R (hunk 8 above)
Comment (jupinator) [reply scripts/protcon_analyisis/protcon_analyisis.R:140]: done

File: pipelines/Protcon-pipeline/Protcon-pipeline.Rmd (hunk 10 above)
Comment (jupinator) [reply pipelines/Protcon-pipeline/Protcon-pipeline.Rmd:30]: done

Comment (jupinator) [reply pipelines/Protcon-pipeline/Protcon-pipeline.Rmd:30]: done

File: pipelines/Protcon-pipeline/Protcon-pipeline.Rmd [hunk 16]
```diff
@@ -0,0 +1,180 @@
...
+  github_document:
+    toc: true             
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+library(knitr)
+knitr::opts_chunk$set(echo = TRUE)
+require("knitr")
+
```
[the remark this answers is not in the document]
Comment (jupinator) [reply pipelines/Protcon-pipeline/Protcon-pipeline.Rmd:13]: done

File: scripts/protcon_analyisis/protcon_analyisis.yml (hunk 4 above)
Comment (jupinator) [reply scripts/protcon_analyisis/protcon_analyisis.yml:71]: there is a better description and example of the protcon results in the markdown file

File: scripts/studyarea_to_WKT/studyarea_to_WKT.yml (hunk 15 above)
[the remark this answers is not in the document]
Comment (jmlord) [reply scripts/studyarea_to_WKT/studyarea_to_WKT.yml:13]: I don't see what was changed here...

File: scripts/studyarea_to_WKT/studyarea_to_WKT.yml (hunk 14 above)
[the remark this answers is not in the document]
Comment (jmlord) [reply scripts/studyarea_to_WKT/studyarea_to_WKT.yml]: In this case, remove the external link. It's not mandatory.

[the remark this answers is not in the document]
Comment (jupinator) [reply scripts/studyarea_to_WKT/studyarea_to_WKT.yml]: I think it is important and I would like to keep it.

File: scripts/protcon_analyisis/protcon_analyisis.yml (hunk 9 above)
Comment (jupinator) [reply scripts/protcon_analyisis/protcon_analyisis.yml]: yes!, sorry and all the calls in others files

File: scripts/spatial_to_DistanceMatrix/spatial_to_DistanceMatrix.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (jmlord) [reply scripts/spatial_to_DistanceMatrix/spatial_to_DistanceMatrix.yml]: Hi @jupinator , I do not see a change? Maybe it was not pushed?

[the remark this answers is not in the document]
Comment (jupinator) [reply scripts/spatial_to_DistanceMatrix/spatial_to_DistanceMatrix.yml]: "String path represents the file location of a spatial file containing spatial units **(geometries)**, which is used to calculate the distance matrix"

File: scripts/studyarea_to_WKT/studyarea_to_WKT.yml (hunk 14 above)
[the remark this answers is not in the document]
Comment (jupinator) [reply scripts/studyarea_to_WKT/studyarea_to_WKT.yml]: i leave this external link: https://github.com/GEO-BON/biab-2.0

File: scripts/studyarea_to_WKT/studyarea_to_WKT.yml (hunk 15 above)
[the remark this answers is not in the document]
Comment (jupinator) [reply scripts/studyarea_to_WKT/studyarea_to_WKT.yml:13]: "Text string representing the file path (from the project root) to the spatial file"...

[the remark this answers is not in the document]
Comment (jupinator) [reply scripts/studyarea_to_WKT/studyarea_to_WKT.yml:13]: "Text string representing the file path (from the project root) to the spatial file containing..."

File: scripts/studyarea_to_WKT/studyarea_to_WKT.yml (hunk 14 above)
[the remark this answers is not in the document]
Comment (jmlord) [reply scripts/studyarea_to_WKT/studyarea_to_WKT.yml:7]: Normally, the point of the external link was to point to something outside of BON in a Box's documentation and metadata.

**Comment by jmlord** [comment]
I'm squash-merging this to a staging branch, where @glaroc will assess that all is good and runs fine, then merge to main.

**Pull request merged**